src/Bicep.Deploy/DeploymentManager.cs

```diff
@@ -19,27 +20,69 @@ public async Task<ArmDeploymentResource> CreateOrUpdateAsync(ArmDeploymentDefini
         {
             var armClient = new ArmClient(new DefaultAzureCredential());
```

Comment:
@shenglol should we be using the credential chain set up in the bicep config? No issues if that's a intentional choice to cut scope for the hackathon.

Reply:
Yes, we should do that. Since the Bicep.Deploy project references Bicep.Core, it already has access to `ITokenCredentialFactory`. @levimatheri here's an example of how we build Azure credentials using Bicep configuration settings: https://github.com/Azure/bicep/blob/e1b781dda84d0baa4760fe19876955f37f7a098e/src/Bicep.LangServer/Providers/AzureContainerRegistriesProvider.cs#L70.